warehouse/accounts/views.py

```diff
@@ -814,3 +818,52 @@ def edit_profile_button(user, request):
 )
 def profile_public_email(user, request):
     return {"user": user}
+
+
+@view_config(
+    require_methods=["POST"],
+    require_csrf=False,
+    renderer="json",
+    route_name="accounts.github-disclose-token",
+    headers=["GITHUB-PUBLIC-KEY-IDENTIFIER", "GITHUB-PUBLIC-KEY-SIGNATURE"],
+    has_translations=True,  # Not the view itself, but the email it sends
+)
+def github_disclose_token(request):
+    # GitHub calls this API view when they have identified a string matching
+    # the regular expressions we provided them.
+    # Our job is to validate we're talking to github, check if the string contains
+    # valid credentials and, if they do, invalidate them and warn the owner
+
+    # The documentation for this process is at
+    # https://developer.github.com/partnerships/token-scanning/
+
+    body = request.body
+
+    # Thanks to the predicates, we know the headers we need are defined.
+    key_id = request.headers.get("GITHUB-PUBLIC-KEY-IDENTIFIER")
+    signature = request.headers.get("GITHUB-PUBLIC-KEY-SIGNATURE")
+
+    verifier = request.find_service(
+        IGitHubTokenScanningPayloadVerifyService, context=None
+    )
+
+    if not verifier.verify(payload=body, key_id=key_id, signature=signature):
+        request.response.status_int = 403
+        return {"error": "invalid signature"}
```

Comment:
I think github does’t care, it fires and forgets.

I’ve tried to put very minimal explainations In the payload to help whoever human person trying to debug this when it stops working someday, but honnestly, empty response would be fine to me, especially if there’s a side method for knowing what’s going on (logs,...)

Reply:
Might be worth checking w/ GitHub to see if they care at all about a response. For example, I'd hope that if they get a 500 they would retry the request, since it's somewhat important that we receive these. Specifically, if this is the case, I'd want to make sure that they consider any 200-level response a success (and not just `200 OK`).

Reply:
(Same, I've asked GitHub, let's wait for their answer)

Reply:
For now I'll go for "empty 200 / empty 400". We'll change this if GitHub gives us more info.